src/fastertransformer/devices/OpData.h

```diff
@@ -551,6 +551,10 @@ struct QuantizeParams {
     OptionalConstBufferRef  static_scale;
     OptionalConstBufferRef  static_scale_reciprocal;
 
+    // for groupwise quantize
+    int64_t    groupSize;
+    QDirection direction;
```

Comment:
Adding this enum in quantize op params is generally not a good idea. a better option is adding a minimal dequantize op. But if this dequantization is for test only, please make it a internal function that do not expose outside rocm implementation. You could dyanmic_cast Device to ROCmDevice, or just make a indepedent function somewhere. Note that you didn't implement this dequantization behaviour for other devices, even nor param check was added, this is inapproperiate.
As a more general rule, do not add anything that not actually related to the framework business logic in device level params.

Reply:
Thanks, understand. 
removed direction member, and add dequant op for rocm only